src/App.js

```diff
@@ -1,30 +1,40 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { useState } from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
 import './App.css';
-import Test from './components/Test';
 import Navigation from './components/Navigation';
 import AddItem from './components/AddItem';
 import List from './components/List';
 import Welcome from './components/Welcome';
+import getToken from './lib/tokens';
 
 function App() {
+  const [token, setToken] = useState(localStorage.getItem('userToken') || '');
+
+  const handleClick = () => {
+    const token = getToken();
+    localStorage.setItem('userToken', token);
+    setToken(token);
+  };
+
   return (
     <div className="App">
       <Switch>
         <Route path="/list">
-          <List />
+          {/* Go to list view if token exists otherwise redirected to home */}
+          {token ? <List token={token} /> : <Redirect to="/" />}
```

Comment:
we're able to successfully pass the user token as props to both the List and AddItem components